homeassistant/components/tplink/select.py

```diff
@@ -0,0 +1,95 @@
+"""Support for TPLink select entities."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Final, cast
+
+from kasa import Device, Feature
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .entity import (
+    CoordinatedTPLinkFeatureEntity,
+    TPLinkDataUpdateCoordinator,
+    TPLinkFeatureEntityDescription,
+    async_refresh_after,
+)
+
+
+@dataclass(frozen=True, kw_only=True)
+class TPLinkSelectEntityDescription(
+    SelectEntityDescription, TPLinkFeatureEntityDescription
+):
+    """Base class for a TPLink feature based sensor entity description."""
+
+
+SELECT_DESCRIPTIONS: Final = [
+    TPLinkSelectEntityDescription(
+        key="light_preset",
+    ),
+    TPLinkSelectEntityDescription(
+        key="alarm_sound",
+    ),
+    TPLinkSelectEntityDescription(
+        key="alarm_volume",
+    ),
+]
+
+SELECT_DESCRIPTIONS_MAP = {desc.key: desc for desc in SELECT_DESCRIPTIONS}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: TPLinkConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors."""
+    data = config_entry.runtime_data
+    parent_coordinator = data.parent_coordinator
+    children_coordinators = data.children_coordinators
+    device = parent_coordinator.device
+
+    entities = CoordinatedTPLinkFeatureEntity.entities_for_device_and_its_children(
+        device=device,
+        coordinator=parent_coordinator,
+        feature_type=Feature.Type.Choice,
+        entity_class=Select,
+        descriptions=SELECT_DESCRIPTIONS_MAP,
+        child_coordinators=children_coordinators,
+    )
+    async_add_entities(entities)
+
+
+class Select(CoordinatedTPLinkFeatureEntity, SelectEntity):
```

Comment:
```suggestion
class TPLinkSelectEntity(CoordinatedTPLinkFeatureEntity, SelectEntity):
```

Reply:
Done in https://github.com/home-assistant/core/pull/120404